streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -744,6 +820,34 @@ private void removeStandbyTasks() {
         standbyRecords.clear();
     }
 
+    private void removeSuspendedTasks() {
+        log.info("{} Removing all suspended tasks [{}]", logPrefix, suspendedTasks.keySet());
+        try {
+            // Close task and state manager
+            for (final AbstractTask task : suspendedTasks.values()) {
+                task.close();
```

Comment:
Also, I think we should also call `task.flushState` since the flushing has been extracted from the close call. And the ordering of these operations should be

1. `task.close()`
2. `task.flushStore()`
3. `task.closeStateManager()`